src/Microsoft.AspNet.Mvc.Core/ViewComponents/IViewComponentSelector.cs

```diff
@@ -1,13 +1,20 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using Microsoft.Framework.Internal;
 
-namespace Microsoft.AspNet.Mvc
+namespace Microsoft.AspNet.Mvc.ViewComponents
 {
+    /// <summary>
+    /// Selects a View Component based on a View Component name.
+    /// </summary>
     public interface IViewComponentSelector
     {
-        Type SelectComponent([NotNull] string componentName);
+        /// <summary>
+        /// Selects a View Component based on <paramref name="componentName"/>.
+        /// </summary>
+        /// <param name="componentName">The View Component name.</param>
+        /// <returns>A <see cref="ViewComponentDescriptor"/>.</returns>
```

Comment:
- worth mentioning `<c>null</c> if no matching View Component is found.`
- or move `throw` here since the caller seems to consistently reject `null`.